tests/constbuffer_array_ut/constbuffer_array_ut.c

```diff
@@ -813,6 +828,556 @@ TEST_FUNCTION(constbuffer_array_create_from_buffer_index_and_count_fails_when_un
+{
+    // arrange
+    CONSTBUFFER_HANDLE test_buffers[3];
+    CONSTBUFFER_ARRAY_HANDLE original;
+    CONSTBUFFER_ARRAY_HANDLE constbuffer_array;
+
+    test_buffers[0] = TEST_CONSTBUFFER_HANDLE_1;
+    test_buffers[1] = TEST_CONSTBUFFER_HANDLE_2;
+    test_buffers[2] = TEST_CONSTBUFFER_HANDLE_3;
+
+    original = constbuffer_array_create(test_buffers, sizeof(test_buffers) / sizeof(test_buffers[0]));
+    umock_c_reset_all_calls();
+
+    // act
+    constbuffer_array = constbuffer_array_create_from_start_and_end(original, 4, 0, 0, 0);
+
+    // assert
+    ASSERT_IS_NULL(constbuffer_array);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    // cleanup
+    constbuffer_array_dec_ref(original);
+}
+
+/* Tests_SRS_CONSTBUFFER_ARRAY_07_002: [ If start_buffer_index is greater than the number of buffers in original then constbuffer_array_create_from_start_and_end shall fail and return NULL. ]*/
+TEST_FUNCTION(constbuffer_array_create_from_start_and_end_with_start_buffer_index_greater_than_buffer_count_for_empty_original_fails)
+{
+    // arrange
+    CONSTBUFFER_ARRAY_HANDLE original;
+    CONSTBUFFER_ARRAY_HANDLE constbuffer_array;
+
+    original = TEST_constbuffer_array_create_empty();
+    umock_c_reset_all_calls();
+
+    // act
+    constbuffer_array = constbuffer_array_create_from_start_and_end(original, 1, 0, 0, 0);
+
+    // assert
+    ASSERT_IS_NULL(constbuffer_array);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    // cleanup
+    constbuffer_array_dec_ref(original);
+}
+
+/* Tests_SRS_CONSTBUFFER_ARRAY_07_003: [ If start_buffer_index + buffer_count is greater than the number of buffers in original then constbuffer_array_create_from_start_and_end shall fail and return NULL. ]*/
+TEST_FUNCTION(constbuffer_array_create_from_start_and_end_with_start_buffer_index_add_buffer_count_greater_than_original_buffer_size_fails)
+{
+    // arrange
+    CONSTBUFFER_HANDLE test_buffers[3];
+    CONSTBUFFER_ARRAY_HANDLE original;
+    CONSTBUFFER_ARRAY_HANDLE constbuffer_array;
+
+    test_buffers[0] = TEST_CONSTBUFFER_HANDLE_1;
+    test_buffers[1] = TEST_CONSTBUFFER_HANDLE_2;
+    test_buffers[2] = TEST_CONSTBUFFER_HANDLE_3;
+
+    original = constbuffer_array_create(test_buffers, sizeof(test_buffers) / sizeof(test_buffers[0]));
+    umock_c_reset_all_calls();
+
+    // act
+    constbuffer_array = constbuffer_array_create_from_start_and_end(original, 1, 3, 0, 0);
+
+    // assert
+    ASSERT_IS_NULL(constbuffer_array);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    // cleanup
+    constbuffer_array_dec_ref(original);
+}
+
+/* Tests_SRS_CONSTBUFFER_ARRAY_07_014: [ If any error occurs then constbuffer_array_create_from_start_and_end shall fail and return NULL. ]*/
+TEST_FUNCTION(constbuffer_array_create_from_start_and_end_fails_when_underlying_functions_fail)

```

Comment:
>constbuffer_array_create_from_start_and_end_fails_when_underlying_functions_fail [](http://example.com/codeflow?start=14&length=80)

since the case for buffer_count=1 is a special case with a different code branch, I think we should also have negative tests for that.

Reply:
nevermind, next test case covers it